src/jit/lclvars.cpp

```diff
@@ -3363,6 +3363,24 @@ void Compiler::lvaAllocOutgoingArgSpace()
         lvaTable[lvaOutgoingArgSpaceVar].lvRefCnt     = 1;
         lvaTable[lvaOutgoingArgSpaceVar].lvRefCntWtd  = BB_UNITY_WEIGHT;
 
+        // For debug IL code compilation we do not create IR for calls to THROW_HELPERs.
+        // The code for the call is generated in codegen (after the frame size is calculated).
+        // If this is the only call in the method, the 4 outgoing arg slots 
+        // will not be allocated.
+        // For release IL compilation an IR block that contains the THROW_HELPER call is always 
+        // generated early enough (in FlowGraph), so the frame size is calculated properly in this case.
+        //
+        // On Windows x64, a frame of a method that has a call must always have 4 slots for 
+        // outgoing arg space - Windows x64 ABI requirement. Make sure these slots are allocated.
+
+        if (compDbgCallsThrowHelper && 
+            compIsAmd64ABI() &&
+            !compIsUnixAmd64ABI() && 
+            (lvaOutgoingArgSpaceSize == 0))
+        {
+            lvaOutgoingArgSpaceSize = 4 * REGSIZE_BYTES;
+        }
+
```

Comment:
Also we should include the profiler sequence below  in the new if then block and remove the  ifdefs of AMD64 and UNIX